doc-content/enterprise-only/optimizer/ex-nQueens-ref.adoc

```diff
@@ -0,0 +1,164 @@
+[id='ex-nQueens-ref']
+= N Queens
+
+Place _n_ queens on a _n_ sized chessboard so that no two queens can attack each other.
+The most common _n_ queens puzzle is the eight queens puzzle, with __n = 8__:
+
+image::UseCasesAndExamples/NQueens/nQueensScreenshot.png[align="center"]
+
+Constraints:
+
+* Use a chessboard of _n_ columns and _n_ rows.
+* Place _n_ queens on the chessboard.
+* No two queens can attack each other. A queen can attack any other queen on the same horizontal, vertical or diagonal line.
+
+This documentation heavily uses the four queens puzzle as the primary example.
+
+A proposed solution could be:
+
+.A Wrong Solution for the Four Queens Puzzle
+image::UseCasesAndExamples/NQueens/partiallySolvedNQueens04Explained.png[align="center"]
+
+The above solution is wrong because queens `A1` and `B0` can attack each other (so can queens `B0` and ``D0``). Removing queen `B0` would respect the "no two queens can attack each other" constraint, but would break the "place _n_ queens" constraint.
+
+Below is a correct solution:
+
+.A Correct Solution for the Four Queens Puzzle
+image::UseCasesAndExamples/NQueens/solvedNQueens04.png[align="center"]
+
+All the constraints have been met, so the solution is correct. 
+
+Note that most _n_ queens puzzles have multiple correct solutions.
+We will focus on finding a single correct solution for a given _n_, not on finding the number of possible correct solutions for a given _n_.
+
+
+[[nQueensProblemSize]]
+.Problem Size
+[source,options="nowrap"]
+----
+4queens   has   4 queens with a search space of    256.
+8queens   has   8 queens with a search space of   10^7.
+16queens  has  16 queens with a search space of  10^19.
+32queens  has  32 queens with a search space of  10^48.
+64queens  has  64 queens with a search space of 10^115.
+256queens has 256 queens with a search space of 10^616.
+----
+
+The implementation of the _n_ queens example has not been optimized because it functions as a beginner example. Nevertheless, it can easily handle 64 queens.
+With a few changes it has been shown to easily handle 5000 queens and more.
+
+
+[[nQueensDomainModel]]
+.Domain Model
+
+This example uses the domain model to solve the four queens problem.
+
+. *Creating a Domain Model* 
```

Comment:
Change these sections to bulleted instead of numbers, and make sure the line breaks are consistent (first doesn't have line break in output, others do).